### PR TITLE
Experimental: in memory deserialization

### PR DESCRIPTION
**Pull request by yoptar**
Done and tested for NER and SQUAD pipelines.
One can serialize all supported chainer components into bytes with
```python3
chainer = build_model(config_path, download=True)
raw_bytes = chainer.serialize()
```
and then deserialize another instance for the same config but without model files with
```python3
chainer = build_model(config_path, serialized=raw_bytes)
```

Every serializable component that does not support this logic will log a warning message that suggests using `load_path`

### CODE REVIEW

File: deeppavlov/models/preprocessors/squad_preprocessor.py [hunk 1]
```diff
@@ -229,8 +229,10 @@ def __init__(self, emb_folder: str, emb_url: str, save_path: str, load_path: str
 
         self.emb_folder.mkdir(parents=True, exist_ok=True)
 
-        if not (self.emb_folder / self.emb_file_name).exists():
-            download(self.emb_folder / self.emb_file_name, self.emb_url)
+        # if not (self.emb_folder / self.emb_file_name).exists():
+        #     download(self.emb_folder / self.emb_file_name, self.emb_url)
```
Comment (nikolay-bushkov) [inline deeppavlov/models/preprocessors/squad_preprocessor.py]: maybe remove this

File: deeppavlov/models/preprocessors/squad_preprocessor.py [hunk 2]
```diff
@@ -299,25 +301,33 @@ def fit(self, contexts: Tuple[List[str], ...], questions: Tuple[List[str]], *arg
...
             self.token2idx_dict[self.OOV] = 1
-            self.embedding_dict[self.NULL] = [0. for _ in range(self.emb_dim)]
-            self.embedding_dict[self.OOV] = [0. for _ in range(self.emb_dim)]
+            self.embedding_dict[self.NULL] = [0.] * self.emb_dim
+            self.embedding_dict[self.OOV] = [0.] * self.emb_dim
             idx2emb_dict = {idx: self.embedding_dict[token]
                             for token, idx in self.token2idx_dict.items()}
             self.emb_mat = np.array([idx2emb_dict[idx] for idx in range(len(idx2emb_dict))])
 
     def load(self, *args, **kwargs):
```
Comment (nikolay-bushkov) [inline deeppavlov/models/preprocessors/squad_preprocessor.py]:
```suggestion
    def load(self, *args, **kwargs) -> None:
```

File: deeppavlov/models/preprocessors/squad_preprocessor.py [hunk 3]
```diff
@@ -299,25 +301,33 @@ def fit(self, contexts: Tuple[List[str], ...], questions: Tuple[List[str]], *arg
...
         logger.info('SquadVocabEmbedder: loading saved {}s vocab from {}'.format(self.level, self.load_path))
-        self.emb_dim, self.emb_mat, self.token2idx_dict = pickle.load(self.load_path.open('rb'))
+        with self.load_path.open('rb') as f:
+            self.emb_dim, self.emb_mat, self.token2idx_dict = pickle.load(f)
+        self.loaded = True
+
+    def deserialize(self, data):
```
Comment (nikolay-bushkov) [inline deeppavlov/models/preprocessors/squad_preprocessor.py]:
```suggestion
    def deserialize(self, data) -> None:
```

File: deeppavlov/models/preprocessors/squad_preprocessor.py [hunk 4]
```diff
@@ -299,25 +301,33 @@ def fit(self, contexts: Tuple[List[str], ...], questions: Tuple[List[str]], *arg
...
+        self.emb_dim, self.emb_mat, self.token2idx_dict = pickle.loads(data)
         self.loaded = True
 
     def save(self, *args, **kwargs):
```
Comment (nikolay-bushkov) [inline deeppavlov/models/preprocessors/squad_preprocessor.py]:
```suggestion
    def save(self, *args, **kwargs) -> None:
```

File: deeppavlov/models/preprocessors/squad_preprocessor.py [hunk 5]
```diff
@@ -299,25 +301,33 @@ def fit(self, contexts: Tuple[List[str], ...], questions: Tuple[List[str]], *arg
...
         logger.info('SquadVocabEmbedder: saving {}s vocab to {}'.format(self.level, self.save_path))
         self.save_path.parent.mkdir(parents=True, exist_ok=True)
-        pickle.dump((self.emb_dim, self.emb_mat, self.token2idx_dict), self.save_path.open('wb'))
+        with self.save_path.open('wb') as f:
+            pickle.dump((self.emb_dim, self.emb_mat, self.token2idx_dict), f)
+
+    def serialize(self):
```
Comment (nikolay-bushkov) [inline deeppavlov/models/preprocessors/squad_preprocessor.py]: @vaskonov what do you think of typing?)

File: deeppavlov/models/embedders/glove_embedder.py [hunk 6]
```diff
@@ -11,6 +11,7 @@
 # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 # See the License for the specific language governing permissions and
 # limitations under the License.
+import pickle
 
 from overrides import overrides
 from typing import Iterator
```
[the remark this answers is not in the document]
Comment (yoptar) [reply deeppavlov/models/embedders/glove_embedder.py]: 👍

File: deeppavlov/core/models/tf_model.py [hunk 7]
```diff
@@ -28,14 +27,16 @@
 
 
 class TFModel(NNModel, metaclass=TfModelMeta):
+    sess: tf.Session
+
```
[the remark this answers is not in the document]
Comment (yoptar) [reply deeppavlov/core/models/tf_model.py]: 👍

**Pull request merged**